transformer_engine/pytorch/attention.py

```diff
@@ -4504,8 +4504,13 @@ def __init__(
         self.attn_mask_type = attn_mask_type
         self.window_size = window_size
         self.window_size = check_set_window_size(attn_mask_type, self.window_size)
-        self.tp_size = tp_size if tp_group is None else get_distributed_world_size(tp_group)
-        self.tp_group = tp_group
+        if tp_group is None:
+            self.tp_size = tp_size
+            if tp_size == 1:
+                self.set_tensor_parallel_group(tp_group)
```

Comment:
I don't think `DotProductAttention` has a `set_tensor_parallel_group()` implementation because it does not derive from `TransformerEngineBaseModule`. Everything else should work once that's implemented.